src/helpers/deprecation.ts

```diff
@@ -0,0 +1,33 @@
+export function deprecateAllQueries<Queries extends Record<string, any>>(
```

Comment:
nit: I find the wording AllQueries confusing since it actually deprecates the queries passed as parameter so I think I'd have named it just `deprecateQueries`, wdyt ?